src/engine/ult.c

```diff
@@ -326,30 +326,27 @@ static inline uint32_t
 sched_ult2xs_multisocket(int xs_type, int tgt_id)
 {
 	static __thread uint32_t offload;
-	uint32_t                 socket;
+	uint32_t                 socket = tgt_id / dss_tgt_per_numa_nr;
 	uint32_t                 base;
 	uint32_t                 target;
 
 	if (dss_tgt_offload_xs_nr == 0) {
 		if (xs_type == DSS_XS_IOFW && dss_forward_neighbor) {
 			/* Keep the old forwarding behavior, but NUMA aware */
-			socket = tgt_id / dss_numa_nr;
 			target = (socket * dss_tgt_per_numa_nr) +
 				 (tgt_id + offload) % dss_tgt_per_numa_nr;
-			offload = target + 17; /* Seed next selection */
-			target  = DSS_MAIN_XS_ID(target);
+			target = DSS_MAIN_XS_ID(target);
 			goto check;
 		}
 		return DSS_XS_SELF;
 	}
 
-	socket  = tgt_id / dss_numa_nr;
-	base    = dss_sys_xs_nr + dss_tgt_nr + (socket * dss_offload_per_numa_nr);
-	target  = base + ((offload + tgt_id) % dss_offload_per_numa_nr);
-	offload = target + 17; /* Seed next selection */
+	base   = dss_sys_xs_nr + dss_tgt_nr + (socket * dss_offload_per_numa_nr);
+	target = base + ((offload + tgt_id) % dss_offload_per_numa_nr);
 
 check:
 	D_ASSERT(target < DSS_XS_NR_TOTAL && target >= dss_sys_xs_nr);
+	offload = target + 17; /* Seed next selection */
```

Comment:
Is this specific to our config? Maybe it should be in a `#define`?

Reply:
Could be 1 probably.   I was trying to make it sort of random as to which ULT it uses next without calling rand